File: my-folder/0053-maximum-subarray/solution.java
class Solution {
    public int maxSubArray(int[] nums) {
      int currMaxSum = nums[0];
      int max = nums[0];
      
      for (int i = 1; i < nums.length; i++) {
        if (currMaxSum > 0) {
          currMaxSum += nums[i];
        } else {
          currMaxSum = nums[i];
        }
        max = Math.max(max, currMaxSum);
      }
      return max;
    }
}

File: my-folder/0053-maximum-subarray/solution.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        n = len(nums)
        if n == 1:
            return nums[0]
        dp = [0]*n
        dp[0] = nums[0]
        for i in range(1, n):
            dp[i] = max(dp[i-1] + nums[i], nums[i])

        ans = -float('inf')
        for i in range(n):
            ans = max(ans, dp[i])
        return ans

